src/meta_schedule/integration.cc

```diff
@@ -121,14 +113,18 @@ ApplyHistoryBest::ApplyHistoryBest(Database database) {
   data_ = n;
 }
 
-Optional<ObjectRef> ApplyHistoryBestNode::Query(runtime::String task_name, IRModule mod,
-                                                Target target,
-                                                Optional<Array<IRModule>> dispatched) {
+IRModule ApplyHistoryBestNode::Query(runtime::String task_name, IRModule mod, Target target,
+                                     Optional<Array<IRModule>> dispatched) {
   ICHECK(dispatched.defined());
   ICHECK_EQ(dispatched.value().size(), 1);
   ICHECK(HasOnlyOneFunction<relay::Function>(mod)) << mod;
   IRModule prim_mod = dispatched.value()[0];
   ICHECK(HasOnlyOneFunction<tir::PrimFunc>(prim_mod)) << prim_mod;
+  // TODO(masahi): parse_mod below replaces the orginal function key with "main".
+  // This is necessary because some scheduling primitives requires the PrimFunc key be "main".
+  // If we can remove this restriction, there would no need for GetOnlyOneFunction* calls below
+  // and we can directly return sch->mod().
+  auto gv = GetOnlyOneFunctionKey<tir::PrimFunc>(prim_mod).value();
```

Comment:
nit: Let be explicit if possible

```suggestion
  GlobalVar gv = GetOnlyOneFunctionKey<tir::PrimFunc>(prim_mod).value();
```